tests/lib/nested.sh

```diff
@@ -687,47 +730,74 @@ start_nested_core_vm(){
         # options, so if that env var is set, we will reuse the existing file if it
         # exists
         IMAGE_NAME="$(get_image_name core)"
-        cp "$IMAGE_DIR/$IMAGE_NAME" "$CURRENT_IMAGE"
+        uncompress_image "$IMAGE_NAME"
+        mv "$NESTED_IMAGES_DIR/$IMAGE_NAME" "$CURRENT_IMAGE"
 
         # Start the nested core vm
         start_nested_core_vm_unit "$CURRENT_IMAGE"
 
         # configure ssh for first time
         prepare_ssh
+
+        # compress the current image if it a generic image
+        if is_generic_image; then
+            force_stop_nested_vm
+            wait_for_service "$NESTED_VM" inactive
+            compress_image "$(get_current_image_name)"
+            force_start_nested_vm
+            wait_for_service "$NESTED_VM"
+            wait_for_ssh
+        fi
     else
         # Start the nested core vm
         start_nested_core_vm_unit "$CURRENT_IMAGE"
-    fi
+    fi    
+}
 
+compress_image() {
+    local IMAGE_NAME=$1
+    if [ ! -f "$NESTED_IMAGES_DIR/$IMAGE_NAME".xz ]; then
+        xz -k0 "$NESTED_IMAGES_DIR/$IMAGE_NAME"
```

Comment:
why compression level 0 here?

Reply:
@anonymouse64, sorry I forgot that question, it is because it is the faster one and all the levels are compressing almost the same size.
At some point I did a comparison between sizes and times for each compression level, I lost the doc with those numbers but what I found is that all the levels were producing the same size but the time was different, the level 0 is the faster and that's why I used level 0 here.
I'll add a comment in a following PR if you think is is appropiate

Reply:
this is fine, no comment needed, that makes sense

Reply:
output:
https://paste.ubuntu.com/p/mRJvGtjdpd/

Reply:
The large image might contain holes so this test is insufficient to determine that.